homeassistant/components/harmony/config_flow.py

```diff
@@ -0,0 +1,187 @@
+from homeassistant.components import ssdp
+from homeassistant.components.remote import (
+    ATTR_ACTIVITY,
+    ATTR_DELAY_SECS,
+    DEFAULT_DELAY_SECS,
+)
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST): str, vol.Required(CONF_NAME): str}, extra=vol.ALLOW_EXTRA
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    harmony = HarmonyAPI(ip_address=data[CONF_HOST])
+
+    _LOGGER.debug("harmony:%s", harmony)
+
+    try:
+        if not await harmony.connect():
+            await harmony.close()
+            raise CannotConnect
+    except harmony_exceptions.TimeOut:
+        raise CannotConnect
+
+    # As a last resort we get the name from the harmony client
+    # in the event a name was not provided
+    if CONF_NAME not in data or data[CONF_NAME] is None or data[CONF_NAME] == "":
+        data[CONF_NAME] = harmony.name
+
+    return {CONF_NAME: data[CONF_NAME], CONF_HOST: data[CONF_HOST]}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Logitech Harmony Hub."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize the Harmony config flow."""
+        self.harmony_config = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            await self.async_set_unique_id(user_input[CONF_HOST])
+            self._abort_if_unique_id_configured()
+            try:
+                info = await validate_input(self.hass, user_input)
+                config_entry = self.async_create_entry(title=info[CONF_NAME], data=info)
+                # Options from yaml are preserved
+                options = _options_from_user_input(user_input)
+                if options:
+                    config_entry["options"] = options
+                return config_entry
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+        # Return form
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a discovered Harmony device."""
+        parsed_url = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION])
+        friendly_name = discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME]
+
+        # The unique id is explicitly not set here because
```

Comment:
You could use context to compare source and logic to allow an import to ignore discovery

Reply:
I dug though the discovery code a bit more and realized there is a concept of a migrated service handler. 

MIGRATED_SERVICE_HANDLERS.  I'm going to remove the discovery from remote.  This will make it a breaking change unfortunately.

Reply:
I'm now checking for unique host and I improved the unique id check